echopype/consolidate/loc_utils.py

```diff
@@ -0,0 +1,173 @@
+            else:
+                potential_datagram_type_set = set(["NMEA", "IDX"])
+            potential_datagram_type_set.discard(datagram_type)
+            for potential_datagram_type in potential_datagram_type_set:
+                # Grab variables from EchoData object
+                if potential_datagram_type != "NMEA":
+                    potential_lat_var = echodata["Platform"].get(
+                        f"latitude_{potential_datagram_type.lower()}"
+                    )
+                    potential_lon_var = echodata["Platform"].get(
+                        f"latitude_{potential_datagram_type.lower()}"
+                    )
+                else:
+                    potential_lat_var = echodata["Platform"]["latitude"]
+                    potential_lon_var = echodata["Platform"]["longitude"]
+
+                # Check if potential variables are present and don't contain all 0s or NaNs
+                if potential_lat_var is not None and potential_lon_var is not None:
+                    potential_contains_zero_lat_lon = (potential_lat_var.values == 0).any() or (
+                        potential_lon_var.values == 0
+                    ).any()
+                    potential_contains_nan_lat_lon = (
+                        np.isnan(potential_lat_var.values).any()
+                        or np.isnan(potential_lon_var.values).any()
+                    )
+
+                    if not (potential_contains_zero_lat_lon or potential_contains_nan_lat_lon):
+                        good_datagram_types.append(potential_datagram_type)
+
+            if len(good_datagram_types) > 0:
+                # Append to shared warning message
+                shared_warning_message = (
+                    shared_warning_message
+                    + f" Consider setting datagram_type to any of {good_datagram_types}."
+                )
+
+        if contains_zero_lat_lon:
+            logger.warning(f"Echodata Platform arrays contain zeros. {shared_warning_message}")
+        if contains_nan_lat_lon:
+            logger.warning(f"Echodata Platform arrays contain NaNs. {shared_warning_message}")
+
+
+def check_loc_time_dim_duplicates(echodata: EchoData, time_dim_name: str) -> None:
+    """Check if there are duplicates in time_dim_name"""
+    if len(np.unique(echodata["Platform"][time_dim_name].data)) != len(
+        echodata["Platform"][time_dim_name].data
+    ):
+        raise ValueError(
+            f'The ``echodata["Platform"]["{time_dim_name}"]`` array contains duplicate values. '
+            "Downstream interpolation on the position variables requires unique time values."
+        )
+
+
+def sel_interp(
+    ds: xr.Dataset, echodata: EchoData, loc_name: str, time_dim_name: str, nmea_sentence: str
+) -> xr.DataArray:
+    """
+    If NMEA sentence is not NaN, selects subset of position variable that is within NMEA sentence,
+    and interpolates position variable to match the ping_time dimension of the input dataset.
```

Comment:
Maybe rewrite this to include the NMEA=NaN case, since that would be the case if people use lat/lon_mru for example.

Reply:
Ah I see. So the 4 cases would be:

1. If `datagram=NMEA` is used and NMEA sentence is NaN, then do nothing.
2. If `datagram=NMEA` is used and NMEA sentence is not NaN, then do the selection.
3. If `datagram!=NMEA` and NMEA sentence is NaN, then do nothing.
4. If `datagram!=NMEA` and NMEA sentence is not NaN, then raise ValueError, ie not allow

I don't think my current code actually deals with the case 4. I'll make this very explicit though.